src/picomsg/src/oper_db_writer.cpp

```diff
@@ -25,6 +25,97 @@ static bool terminator_state_to_oper_db(const pm_terminator_conn_state terminato
 	return false;
 }
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wswitch-enum"
+static bool tunnel_status_to_oper_db(ike_sa_state_t tunnel_status, enum LIBOPER_DB_CLIENT_FIELD_ENUM_VALUES& oper_db_value) {
+	switch(tunnel_status) {
+		case IKE_CREATED:
+			oper_db_value = LIBOPER_DB_CLIENT_STATUS_IKE_CREATED;
+			return true;
+		case IKE_CONNECTING:
+			oper_db_value = LIBOPER_DB_CLIENT_STATUS_IKE_CONNECTING;
+			return true;
+		case IKE_ESTABLISHED:
+			oper_db_value = LIBOPER_DB_CLIENT_STATUS_IKE_ESTABLISHED;
+			return true;
+		case IKE_REKEYING:
+			oper_db_value = LIBOPER_DB_CLIENT_STATUS_IKE_REKEYING;
+			return true;
+		case IKE_REKEYED:
+			oper_db_value = LIBOPER_DB_CLIENT_STATUS_IKE_REKEYED;
+			return true;
+		case IKE_DELETING:
+			oper_db_value = LIBOPER_DB_CLIENT_STATUS_IKE_DELETING;
+			return true;
+		default:
+			PICOMSG_LOG(PICOMSG_LOG_ERR, "got invalid ike state %d", tunnel_status);
+			return false;
+	}
+
+	return false;
+}
+
+static bool version_to_oper_db(const uint32_t version, enum LIBOPER_DB_CLIENT_FIELD_ENUM_VALUES& oper_db_value) {
+	switch(version) {
+		case 1:
+				oper_db_value = LIBOPER_DB_CLIENT_IKE_VERSION_1;
+				return true;
+		case 2:
+				oper_db_value = LIBOPER_DB_CLIENT_IKE_VERSION_2;
+				return true;
+	}
+	PICOMSG_LOG(PICOMSG_LOG_ERR, "got invalid ike version %u", version);
+	return false;
+}
+
+static bool activeness_status_to_oper_db(const bool status, enum LIBOPER_DB_CLIENT_FIELD_ENUM_VALUES& oper_db_value) {
+	switch(status) {
```

Comment:
`status` is a boolean. You don't need a switch-case for a boolean.. (and even if in some world it was appropriate, use at least `false` and `true` as the values instead of `0` and `1`)